src/ACEtk/block/details/BaseDistributionData.hpp

```diff
@@ -0,0 +1,193 @@
+
+  /**
+   *  @brief Return the number of interpolation regions
+   */
+  std::size_t numberInterpolationRegions() const { return this->NB(); }
+
+  /**
+   *  @brief Return the interpolation boundaries
+   */
+  auto NBT() const { return this->interpolationData().NBT(); }
+
+  /**
+   *  @brief Return the interpolation boundaries
+   */
+  auto boundaries() const { return this->NBT(); }
+
+  /**
+   *  @brief Return the interpolants
+   */
+  auto INT() const { return this->interpolationData().INT(); }
+
+  /**
+   *  @brief Return the interpolants
+   */
+  auto interpolants() const { return this->INT(); }
+
+  /**
+   *  @brief Return the number of incident energy values
+   */
+  std::size_t NE() const { return this->data_.N(); }
+
+  /**
+   *  @brief Return the number of incident energy values
+   */
+  std::size_t numberIncidentEnergies() const { return this->NE(); }
+
+  /**
+   *  @brief Return the incident energy values
+   */
+  auto incidentEnergies() const { return this->data_.column( 1 ); }
+
+  /**
+   *  @brief Return the incident energy for a given index
+   *
+   *  When the index is out of range an std::out_of_range exception is thrown
+   *  (debug mode only).
+   *
+   *  @param[in] index     the index (one-based)
+   */
+  double incidentEnergy( std::size_t index ) const {
+
+    #ifndef NDEBUG
+    this->verifyIncidentEnergyIndex( index );
+    #endif
+    return this->data_.value( 1, index );
+  }
+
+  /**
+   *  @brief Return the angular distribution locator for an incident
+   *         energy index
+   *
+   *  This locator is the value as stored in the XSS array.
+   *
+   *  When the index is out of range an std::out_of_range exception is thrown
+   *  (debug mode only).
+   *
+   *  @param[in] index     the index (one-based)
+   */
+  int LOCC( std::size_t index ) const {
+
+    #ifndef NDEBUG
+    this->verifyIncidentEnergyIndex( index );
+    #endif
+    return this->data_.value( 2, index );
+  }
+
+  /**
+   *  @brief Return the angular distribution locator for an incident
+   *         energy index
+   *
+   *  This locator is the value as stored in the XSS array.
+   *
+   *  When the index is out of range an std::out_of_range exception is thrown
+   *  (debug mode only).
+   *
+   *  @param[in] index     the index (one-based)
+   */
+  int distributionLocator( std::size_t index ) const {
+
+    return this->LOCC( index );
+  }
+
+  /**
+   *  @brief Return the relative angular distribution locator for an incident
+   *         energy index
+   *
+   *  This is the locator relative to the beginning of the current distribution
+   *  block.
+   *
+   *  When the index is out of range an std::out_of_range exception is thrown
+   *  (debug mode only).
+   *
+   *  @param[in] index     the index (one-based)
+   */
+  std::size_t relativeDistributionLocator( std::size_t index ) const {
+
+    const int locator = this->LOCC( index );
+    return locator - this->locb_ + 1;
```

Comment:
[design question]
The locator is an integer, yet we're returning a size_t. Is there a reason for the conversion?

Reply:
Me sloppy with integer types.

Reply:
Changes in fix/review-part1